src/cpp/fastdds/subscriber/history/DataReaderHistory.hpp

```diff
@@ -214,6 +225,9 @@ class DataReaderHistory : public eprosima::fastrtps::rtps::ReaderHistory
     void check_and_remove_instance(
             instance_info& instance_info);
 
+    /// Function to compute the instance handle of a received change
+    std::function<bool(CacheChange_t*)> compute_key_for_change_fn;
```

Comment:
Me neither. I did the change in a previous implementation and forget to put back in place.

Reply:
Now it is used by `DataReaderImpl::update_smaple_rejected_status()`.

Reply:
I think it would be nicer to have a public method `bool compute_key_for_change(CacheChange_t*)`, and leave the `compute_key_for_change_fn` private. This way, we prevent `compute_key_for_change_fn` to be assigned externally.

Reply:
Either that or not using it from `update_smaple_rejected_status`, and allow (and document) sample_rejected_status to have an unknown instance_handle.

Reply:
Done in 97d98